Microsoft.Toolkit.Uwp.UI.Controls/RemoteDevicePicker/RemoteDevicePicker.cs

```diff
@@ -0,0 +1,210 @@
+            if (accessStatus == RemoteSystemAccessStatus.Allowed)
+            {
+                RemoteSystemWatcher m_remoteSystemWatcher = RemoteSystem.CreateWatcher();
+                m_remoteSystemWatcher.RemoteSystemAdded += RemoteSystemWatcher_RemoteSystemAdded;
+                m_remoteSystemWatcher.RemoteSystemRemoved += RemoteSystemWatcher_RemoteSystemRemoved;
+                m_remoteSystemWatcher.RemoteSystemUpdated += RemoteSystemWatcher_RemoteSystemUpdated;
+                m_remoteSystemWatcher.Start();
+            }
+
+            _progressRing.IsActive = true;
+            UpdateList();
+        }
+
+        private async void RemoteSystemWatcher_RemoteSystemUpdated(RemoteSystemWatcher sender, RemoteSystemUpdatedEventArgs args)
+        {
+            await Dispatcher.RunAsync(CoreDispatcherPriority.Normal, () =>
+            {
+                _progressRing.IsActive = true;
+                if (DeviceMap.ContainsKey(args.RemoteSystem.Id))
+                {
+                    RemoteSystems.Remove(DeviceMap[args.RemoteSystem.Id]);
+                    DeviceMap.Remove(args.RemoteSystem.Id);
+                }
+                RemoteSystems.Add(args.RemoteSystem);
+                DeviceMap.Add(args.RemoteSystem.Id, args.RemoteSystem);
+                UpdateList();
+            });
+        }
+
+        internal void UpdateList()
+        {
+            ObservableCollection<RemoteSystem> bindingList = new ObservableCollection<RemoteSystem>();
+            if (RemoteSystems != null)
+            {
+                foreach (RemoteSystem sys in RemoteSystems)
+                {
+                    if (_listDeviceTypes.SelectedValue.ToString().Equals(DeviceType.All.ToString()))
+                    {
+                        bindingList = RemoteSystems;
+                    }
+                    else if (_listDeviceTypes.SelectedValue.ToString().Equals(sys.Kind))
+                    {
+                        bindingList.Add(sys);
+                    }
+                }
+
+                _progressRing.IsActive = false;
+            }
+
+            _listDevices.ItemsSource = bindingList;
+        }
+
+        private async void RemoteSystemWatcher_RemoteSystemRemoved(RemoteSystemWatcher sender, RemoteSystemRemovedEventArgs args)
+        {
+            await Dispatcher.RunAsync(CoreDispatcherPriority.Normal, () =>
+            {
+                _progressRing.IsActive = true;
+                if (DeviceMap.ContainsKey(args.RemoteSystemId))
+                {
+                    RemoteSystems.Remove(DeviceMap[args.RemoteSystemId]);
+                    DeviceMap.Remove(args.RemoteSystemId);
+                }
+            });
+        }
+
+        private async void RemoteSystemWatcher_RemoteSystemAdded(RemoteSystemWatcher sender, RemoteSystemAddedEventArgs args)
+        {
+            await Dispatcher.RunAsync(CoreDispatcherPriority.Normal, () =>
```

Comment:
I thought we still have `DispatcherHelper.ExecuteOnUIThreadAsync` ? What are all of the others controls using these days for the toolkit?